src/System.Windows.Forms/src/System/Windows/Forms/HScrollProperties.cs

```diff
@@ -13,16 +11,16 @@ namespace System.Windows.Forms
     /// </summary>
     public class HScrollProperties : ScrollProperties
     {
-        public HScrollProperties(ScrollableControl container) : base(container)
+        public HScrollProperties(ScrollableControl? container) : base(container)
         {
         }
 
-        internal override int PageSize => ParentControl.ClientRectangle.Width;
+        internal override int PageSize => ParentControl!.ClientRectangle.Width;
```

Comment:
Are you sure this is correct?

We can pass `null` into the ctor, which is then passed into `base`.
`ScrollProperties` does not validate and guarantee `ParentControl` not be `null`.

It appears the implementations of `HScrollProperties` and `VScrollProperties` suffer the same symptoms incorrectly assuming `ParentControl != null`.
I think we should correct the implementation to:
```suggestion
        internal override int PageSize => ParentControl?.ClientRectangle.Width ?? 0;
```

Reply:
This property is called from `ScrollProperties.UpdateScrollInfo`, which has an explicit check for the nullability of ParentControl. See

```
internal void UpdateScrollInfo()
{
    if (_parent != null && _parent.IsHandleCreated && _visible)
    {
        var si = new User32.SCROLLINFO
        {
            cbSize = (uint)Marshal.SizeOf<User32.SCROLLINFO>(),
            fMask = User32.SIF.ALL,
            nMin = _minimum,
            nMax = _maximum,
            nPage = _parent.AutoScroll ? (uint)PageSize : (uint)LargeChange,
            nPos = _value,
            nTrackPos = 0
        };
        User32.SetScrollInfo(_parent, Orientation, ref si, BOOL.TRUE);
    }
}
```

I've improved this code by making these properties `private protected`

Reply:
> This property is called from ScrollProperties.UpdateScrollInfo, which has an explicit check

The debugger may evaluate properties without doing any additional checks. Also relying on the caller doing checks before accessing a property is fragile design, future callers may forget the check. (I'm not arguing for changing it, just adding some additional scenarios to consider.)